cpp/tests/io/parquet_test.cpp

```diff
@@ -866,10 +866,10 @@ TEST_F(ParquetChunkedWriterTest, SimpleTable)
   auto filepath = temp_env->get_temp_filepath("ChunkedSimple.parquet");
   cudf_io::chunked_parquet_writer_options args =
     cudf_io::chunked_parquet_writer_options::builder(cudf_io::sink_info{filepath});
-  auto state = cudf_io::write_parquet_chunked_begin(args);
-  cudf_io::write_parquet_chunked(*table1, state);
-  cudf_io::write_parquet_chunked(*table2, state);
-  cudf_io::write_parquet_chunked_end(state);
+  cudf_io::parquet_chunked_writer writer(args);
+  writer.write(*table1);
+  writer.write(*table2);
+  writer.write_end();
```

Comment:
I have an idea about this API. How about we return *this from write? This call site would become:
```
cudf_io::parquet_chunked_writer(args).write(*table1).write(*table2);
```
Since the writer is a temporary, dtor is called immediately.
If output metadata is required, users can do something like:
```
auto const md = cudf_io::parquet_chunked_writer(args).write(*table1).write(*table2).write_end(true, xyz);
```
IMO it's an improvement because users don't need a variable in some cases - no variable, no way to call `write` after `write_end` :)
Thoughts?

Reply:
They should return reference,  it would be nice and will provide both option.